File: app/src/main/java/com/aquaguard/domain/repository/WaterRepository.java
package com.aquaguard.domain.repository;

import androidx.lifecycle.LiveData;
import com.aquaguard.domain.model.WaterReading;
import java.util.Map;

public interface WaterRepository {
    LiveData<WaterReading> getLiveReading(String deviceId);
    void toggleValve(String deviceId, boolean open, String triggeredBy, String reason, RepositoryCallback<Void> callback);
    void toggleAutoMode(String deviceId, boolean enabled, RepositoryCallback<Void> callback);
    LiveData<Map<String, Float>> getDailyUsage(String deviceId);
    LiveData<Map<String, Float>> getMonthlyUsage(String deviceId);
    void recordUsage(String deviceId, float liters, float waterSaved, RepositoryCallback<Void> callback);
}

File: Aqua Guard/frontend/app/src/main/java/com/aquaguard/domain/repository/WaterRepository.kt
package com.aquaguard.domain.repository

import com.aquaguard.domain.model.WaterReading
import kotlinx.coroutines.flow.Flow

interface WaterRepository {
    fun getLiveReading(deviceId: String): Flow<WaterReading?>
    suspend fun toggleValve(deviceId: String, open: Boolean, triggeredBy: String, reason: String): Result<Unit>
    suspend fun toggleAutoMode(deviceId: String, enabled: Boolean): Result<Unit>
    fun getDailyUsage(deviceId: String): Flow<Map<String, Float>>
    fun getMonthlyUsage(deviceId: String): Flow<Map<String, Float>>
    suspend fun recordUsage(deviceId: String, liters: Float, waterSaved: Float): Result<Unit>
}
